service/datasource-service.py

```diff
@@ -155,12 +162,52 @@ def decorated(*args, **kwargs):
 
     return decorated
 
+def generate_sheetdata(url,auth,params,headers,sheets,ids,names,direction,start,since):
+    logger.info("downloading from url=%s with params=%s" % (url, params))
+    r = requests.get(url, auth=auth, headers=headers, params=params)
+    r.raise_for_status()
+    logger.debug("Got file data")
+    workbook = xlrd.open_workbook("sesm.xsls", file_contents=r.content)
+    if workbook.props["modified"] > since:
+        iterable_sheets = sheets or range(0,Workbook.nsheets)
+        for sheet_index in iterable_sheets:
+            worksheet = workbook.sheet_by_index(sheet_index)
+            id_prefix = str(sheet_index+1) + "-" if len(iterable_sheets)>1 else ""
+            if direction == "row":
+                columnNames = getColNames(worksheet, names, start)
+                yield from getSheetRowData(worksheet, columnNames, start, ids, workbook.props["modified"], workbook.datemode, id_prefix)
+            else:
+                rowNames = getRowNames(worksheet, names, start)
+                yield from getSheetColData(worksheet, rowNames, start, ids, workbook.props["modified"], workbook.datemode, id_prefix)
 
-@app.route('/')
+@app.route('/', methods=["GET"])
+@app.route('/<path:path>', methods=["GET"])
 @requires_auth
-def get_entities():
-    file = get_var('file')
-    sheet = int(get_var('sheet') or 1) - 1
+def get_entities(path=None):
+    url = get_var('file')
+    file = url
+    logger.debug(f'path={path}')
+    headers, params, auth = None , None, None
+    if not url:
```

Comment:
here support for reusability is being added. instead of expecting exact file url, we parse a new optional envvar _download_request_spec_ which contains base url. Then we append the path variable to construct the final url.